flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -252,10 +274,14 @@ public void onContainersAllocated(List<Container> containers) {
 				// failed to launch the container, will release the failed one and ask for a new one
 				LOG.error("Could not start TaskManager in container {},", container, t);
 				resourceManagerClient.releaseAssignedContainer(container.getId());
-				requestYarnContainer(container.getResource(), container.getPriority());
+				requestYarnContainer(container.getResource(), priority);
 			}
 		}
-		if (numPendingContainerRequests <= 0) {
+		int pendingRequest = 0;
+		for (int num : numPendingContainerRequests.values()) {
+			pendingRequest += num;
+		}
```

Comment:
This does not look efficient. Why not having a global counter across all different requests?